middle_end/flambda2.0/types/template/flambda_type.templ.ml

```diff
@@ -315,7 +316,11 @@ let prove_tags_must_be_a_block env t : Tag.Set.t proof =
             if Tag.Set.is_empty tags then Invalid
             else Proved tags
     end
-  | Value (Ok _) -> Invalid
+  | Value (Ok (Boxed_float _ | Boxed_int32 _ | Boxed_int64 _
+      | Boxed_nativeint _)) -> Proved (Tag.Set.singleton Tag.custom_tag)
+  | Value (Ok (Closures _)) -> Proved (Tag.Set.singleton Tag.closure_tag)
```

Comment:
Shouldn't this include `Tag.infix_tag` ?

Reply:
Fixed